ingestJson.py

```diff
@@ -1,51 +1,84 @@
+#!/usr/bin/env python
+
+import argparse
 import requests
 import json
-import sys
-
-
-def help():
-    print
-    print "ingestJson ingests API data from mantid production to local server."
-    print "To use, provide a mantid API name as an argument."
-    print "usage: python ingestJson.py <apiName>"
-    print "example: python ingestJson.py feature"
-    print "would pull from mantidproject.org/api/feature and post @ localhost."
-    exit(0)
 
 
-def receive_json():
-    print "Pulling data from " + get_url
-    r = requests.get(get_url, params={'format': 'json'})
+def receive_json(source, page_num):
+    if args.verbose:
+        print "Pulling data from " + source
+    r = requests.get(source, params={'page': str(page_num), 'format': 'json'})
+    if args.verbose:
+        print "Received", r.content.__sizeof__(), "bytes from source."
     return r.content
 
 
-def send(data):
-    print "Posting to " + post_url + "\n"
-    r = requests.post(post_url, headers={
+def post_json(data, destination):
+    r = requests.post(destination, headers={
                       "Content-Type": "application/json"}, data=data)
-    if str(r) == "<Response [201]>":
-        print "OK. Returned an HTTP 201 for Created."
+    errors = 0
+    if r.status_code == 201:
+        # if args.verbose:
+        #    print "OK. Returned an HTTP 201 for Created."
+        pass
     else:
-        print "Response:"
-        print r
+        # if args.verbose:
+        #    print "Error. Response:", r.status_code
+        errors += 1
+    return errors
 
 
-def main():
-    apiData = receive_json()
-    true_json = json.loads(apiData)
+def iterate_and_post(jsonData, destination):
+    try:
+        true_json = json.loads(jsonData)
+    except:
+        print "Failed to load JSON data."
+        print "Have you tried removing slashes from the end of the URL?"
+        return 1
+    if args.verbose:
+        print "Posting objects to", destination
+    i = 0
+    errors = 0
     for element in true_json["results"]:  # iterates over inner objects
-        send(json.dumps(element))  # removes unicode
+        # json.dumps removes unicode
+        errors += post_json(json.dumps(element), destination)
+        i += 1
+    if errors == 0:
+        if args.verbose:
+            print "Posted", i, "JSON objects to destination."
+    else:
+        if args.verbose:
+            print "Attempted", i, "JSON posts, resulted with", errors, "errors."
+    return errors
+
 
+parser = argparse.ArgumentParser()
```

Comment:
Add an argument (with options) to pull from specific api names. It will direct the user to the correct endpoints and you can error check the input arguments more.

Move setting up the parser inside `__name__ == '__main__'`